source/frontend/src/features/mapSideBar/disposition/models/DispositionSaleFormModel.ts

```diff
@@ -38,6 +52,20 @@ export class DispositionSaleFormModel {
 
     model.netProceedsBeforeSppAmount = calculateNetProceedsBeforeSppAmount(entity);
     model.netProceedsAfterSppAmount = calculateNetProceedsAfterSppAmount(entity);
+
+    model.dispositionPurchasers = entity.dispositionPurchasers?.map(
+      x => DispositionSaleContactModel.fromApi(x) || [],
+    );
+
+    model.dispositionPurchaserAgent = entity.dispositionPurchaserAgent
+      ? DispositionSaleContactModel.fromApi(entity.dispositionPurchaserAgent)
+      : new DispositionSaleContactModel(null, entity.id, 0, null);
```

Comment:
this constructor is being called with the same hardcoded parameters, that look like a codesmell. instead how about using a static method that builds the object using only the entity id?

Reply:
I would rather just simplify the constructor instead of adding another method.